bsp/Inc/BSP_SPI.h

```diff
@@ -0,0 +1,61 @@
+#ifndef BSP_SPI_H
+#define BSP_SPI_H
+
+#include <stdint.h>
+#include "stm32f4xx.h"
+#include "stm32f4xx_hal.h"
+#include "stm32f4xx_hal_def.h"
+#include "stm32f4xx_hal_gpio.h"
+#include "stm32f4xx_hal_spi.h"
+#include "stm32f4xx_hal_rcc.h"
+#include "FreeRTOS.h"
+#include "semphr.h"
+
+// SPI prescalers
+typedef enum {
+    SLOW,
+    FAST,
+    NUM_SPEEDS
+} BSP_SPI_Speed;
+static const uint32_t _speedLut[NUM_SPEEDS] = {SPI_BAUDRATEPRESCALER_256, SPI_BAUDRATEPRESCALER_128};
```

Comment:
Do we really need to predefine these for the user? Also I agree with lakshay, we can set it in the init

Reply:
these were pulled from the original SPI files, we don't need them necessarily